### PR TITLE
4629 - Add ids checkbox and more

### PR DESCRIPTION
**Pull request by deep7102**
**Explain the details for making this change. What existing problem does the pull request solve?**
- ids-checkbox: add checkbox component
- ids-hide-focus-mixin: handle focus/click events
- ids-dirty-tracker-mixin: add support for checkbox
- ids-validation-mixin: add support to handle multi events and support for checkbox
- ids-input: Remove attribute `name/fieldName` was in conflict
- Added docs and tests
- Added standalone css example

**Related github/jira issue (required)**:
Fixes infor-design/enterprise#4629

**Steps necessary to review your pull request (required)**:
- Pull this branch and build/run the demo app
- Navigate to:
- http://localhost:4300
- http://localhost:4300/ids-checkbox
- http://localhost:4300/ids-checkbox/standalone-css.html
- See checkbox uplift styles
- See checked, unchecked, disabled checkbox
- See checkbox colors
- Test dirty tracker
- See required label with red `*` star and no-indicator without red `*` star
- Test validation for required (must be checked to pass validation)
- Test to set/remove indeterminate state (click on checkbox will change state to checked/unchecked)

**Included in this Pull Request**:
- [x] An e2e or functional test for the bug or feature.

### CODE REVIEW

**Review by tmcconechy — CHANGES_REQUESTED** [review]
Looking really good 🍌 Just a few suggestions inline and as well:

- can you check the branch coverage its only 93.1% for checkbox i think that can be fixed. (and any others you want).
- for standalone really just need the first standard parts - but since you did it dont have to take it out
- in terms of styling the only thing i see is a bit too much margin between the checkboxes . Should be 16px now but its closer to 20
- e2e tests?

Also assume the radios and switch will follow? Then maybe dont close the issue on the pr.

File: app/ids-checkbox/example.html [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      <span slot="text">Set</span>
+    </ids-button>
+    <ids-button type="secondary" id="btn-remove-indeterminate">
+      <span slot="text">Remove</span>
+    </ids-button>
+
+    <br/><br/>
+    <ids-text font-size="12" type="h1">Ids Checkbox - Horizontal</ids-text>
+
+    <ids-checkbox label="Option 1 I am long" display-inline="true"></ids-checkbox>
```
[the remark this answers is not in the document]
Comment (deep7102) [reply app/ids-checkbox/example.html]: Update now as horizontal

File: src/ids-base/ids-colorpalette-variables.scss [hunk 2]
```diff
@@ -0,0 +1,23 @@
+// Color Variables
+//================================================== //
+$color-status-names: info caution danger success warning;
+
+$color-names: ruby01 ruby02 ruby03 ruby04 ruby05 ruby06 ruby07 ruby08 ruby09 ruby10
```
[the remark this answers is not in the document]
Comment (deep7102) [reply src/ids-base/ids-colorpalette-variables.scss:3]: Updated now, use as css variables

File: src/ids-checkbox/README.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
+# Ids Checkbox Component
+
+## Description
+
+The IDS Checkbox component is a simple wrapper around a standard HTMLInputElement that is styled with Infor branding, and contains some additional API that makes it easy to set checkbox, label, and functionality. IDS Checkbox let user select one or more values at a time.
```
[the remark this answers is not in the document]
Comment (deep7102) [reply src/ids-checkbox/README.md]: updated now

File: src/ids-checkbox/README.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+```html
+<!-- 4.x input example -->
+<div class="field">
+  <input type="checkbox" class="checkbox" name="checkbox1" id="checkbox1"/>
+  <label for="checkbox1" class="checkbox-label">Unchecked</label>
+</div>
+
+<!-- this is the same input using the WebComponent -->
+<ids-checkbox label="Unchecked"></ids-checkbox>
+
```
[the remark this answers is not in the document]
Comment (deep7102) [reply src/ids-checkbox/README.md]: updated now

**Comment by deep7102** [comment]
Hi @tmcconechy updated now, the points you found above.

**Review by tmcconechy — APPROVED** [review]

**Review by EdwardCoyle — APPROVED** [review]

**Pull request merged**